tests/system/python/packages/test_pi_webapi.py

```diff
@@ -128,58 +168,99 @@ def test_end_to_end(self, start_south_north, read_data_from_pi_web_api, fledge_u
+            shell=True, check=True)
+
+        time.sleep(wait_time)
 
-            assert 1 == sent, "Failed to send data via PI Web API using Basic auth"
+        verify_ping(fledge_url, skip_verify_north_interface, wait_time, retries)
+        verify_asset(fledge_url)
+        verify_statistics_map(fledge_url, skip_verify_north_interface)
+        verify_asset_tracking_details(fledge_url, skip_verify_north_interface)
 
-        actual_stats_map = get_statistics_map(fledge_url)
-        assert 1 == actual_stats_map[asset_name.upper()]
-        assert 1 == actual_stats_map['READINGS']
+        # Good reconfiguration to check data is sent
+        data = {"SendFullStructure": "false"}
+        put_url = "/fledge/category/{}".format(north_service_name)
+        resp = utils.put_request(fledge_url, urllib.parse.quote(put_url), data)
+        assert "false" == resp["SendFullStructure"]["value"]
+
+        old_ping_result = verify_ping(fledge_url, skip_verify_north_interface, wait_time, retries)
+
+        conn = http.client.HTTPConnection(fledge_url)
+        subprocess.run(
+            ["cd $FLEDGE_ROOT/extras/python; python3 -m fogbench -t ../../data/{}; cd -".format(TEMPLATE_NAME)],
+            shell=True, check=True)
+
+        # Wait for the OMF schedule to run.
+        time.sleep(wait_time * 2)
+
+        new_ping_result = verify_ping(fledge_url, skip_verify_north_interface, wait_time, retries)
+        # Verifies whether Read and Sent readings are increasing after delete/add of north service
+        assert old_ping_result['dataRead'] < new_ping_result['dataRead']
         if not skip_verify_north_interface:
-            assert 1 == actual_stats_map['Readings Sent']
-            assert 1 == actual_stats_map['NorthReadingsToPI_WebAPI']
-
-        conn.request("GET", '/fledge/asset')
-        r = conn.getresponse()
-        assert 200 == r.status
-        r = r.read().decode()
-        retval = json.loads(r)
-        assert len(retval) == 1
-        assert asset_name == retval[0]["assetCode"]
-        assert 1 == retval[0]["count"]
-
-        conn.request("GET", '/fledge/asset/{}'.format(asset_name))
-        r = conn.getresponse()
-        assert 200 == r.status
-        r = r.read().decode()
-        retval = json.loads(r)
-        assert {DATAPOINT: DATAPOINT_VALUE} == retval[0]["reading"]
+            assert old_ping_result['dataSent'] < new_ping_result['dataSent']
+
+        # Bad reconfiguration to check data is not sent
+        data = {"PIWebAPIUserId": "Admin"}
+        put_url = "/fledge/category/{}".format(north_service_name)
+        resp = utils.put_request(fledge_url, urllib.parse.quote(put_url), data)
+        assert "Admin" == resp["PIWebAPIUserId"]["value"]
+
+        old_ping_result = verify_ping(fledge_url, skip_verify_north_interface, wait_time, retries)
+
+        # Wait for the OMF schedule to run and then send the new data.
+        # Otherwise, it sends the data with old config. 
```

Comment:
not clear what do you mean `Otherwise, it sends the data`? as you have configured with bad value it will not be sent anyway for this run.

I understand we need the next task iteration schedule to run hence the wait (default 15s hence 15s+ wait)?

Reply:
that sounds like a bug to me. if Re-config is successful, the immediate next task schedule should respect that. May be scheduler has already queued.